homeassistant/components/nextcloud/sensor.py

```diff
@@ -0,0 +1,190 @@
+"""Summary data from Nextcoud."""
+from datetime import timedelta
+import logging
+
+from nextcloudmonitor import NextcloudMonitor
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_URL,
+    CONF_USERNAME,
+)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+# Assign domain
+DOMAIN = "nextcloud"
+# Set Default scan interval in seconds
+SCAN_INTERVAL = timedelta(seconds=60)
+
+# Validate user configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+    }
+)
+
+# Add expected items and paths from nextcloud monitor api
+NEXTCLOUD_ITEMS_EXPECTED = {
```

Comment:
All expected items should and all communication to external devices should be moved to the python package. See the [development checklist](https://developers.home-assistant.io/docs/en/development_checklist.html) for reference.

Reply:
@springstan  I see where the documentation asks that `All communication to external devices or services must be wrapped in an external Python library hosted on pypi.` but I'm not seeing how that relates to the expected items. The pypi package's purpose is to return all the data that Nextcloud provides. This integration only reports a subset of that data. 

I'll see if i can re-work the integration to dynamically determine the pieces of information to use as sensors.

Reply:
Yeah I was thinking that the pypi package could only return the pieces of information to use as sensors that are available.